diff -up webkit-qtwebkit/WebCore/WebCore.pro.phonon webkit-qtwebkit/WebCore/WebCore.pro
--- webkit-qtwebkit/WebCore/WebCore.pro.phonon	2011-04-15 09:32:17.000000000 -0500
+++ webkit-qtwebkit/WebCore/WebCore.pro	2011-04-21 08:42:17.519666953 -0500
@@ -2571,14 +2571,14 @@ contains(DEFINES, ENABLE_VIDEO=1) {
         rendering/RenderMedia.cpp \
         bindings/js/JSAudioConstructor.cpp
 
-        !lessThan(QT_MINOR_VERSION, 6):contains(MOBILITY_CONFIG, multimedia) {
-            HEADERS += platform/graphics/qt/MediaPlayerPrivateQt.h
-            SOURCES += platform/graphics/qt/MediaPlayerPrivateQt.cpp
-
-            CONFIG *= mobility
-            MOBILITY += multimedia
-            DEFINES += WTF_USE_QT_MULTIMEDIA
-         } else:contains(QT_CONFIG, phonon) {
+#        !lessThan(QT_MINOR_VERSION, 6):contains(MOBILITY_CONFIG, multimedia) {
+#            HEADERS += platform/graphics/qt/MediaPlayerPrivateQt.h
+#            SOURCES += platform/graphics/qt/MediaPlayerPrivateQt.cpp
+#
+#            CONFIG *= mobility
+#            MOBILITY += multimedia
+#            DEFINES += WTF_USE_QT_MULTIMEDIA
+#         } else:contains(QT_CONFIG, phonon) {
             HEADERS += \
                 platform/graphics/qt/MediaPlayerPrivatePhonon.h
 
@@ -2595,7 +2595,7 @@ contains(DEFINES, ENABLE_VIDEO=1) {
                 INCLUDEPATH -= $$QMAKE_LIBDIR_QT/phonon.framework/Headers
                 INCLUDEPATH += $$QMAKE_LIBDIR_QT/phonon.framework/Headers
             }
-        }
+#        }
 }
 
 contains(DEFINES, ENABLE_XPATH=1) {
